src/SFA.DAS.ApplyService.Application/Apply/Oversight/RecordOversightGatewayFailOutcomeCommandHandler.cs

```diff
@@ -31,6 +32,13 @@ public async Task<Unit> Handle(RecordOversightGatewayFailOutcomeCommand request,
         {
             _logger.LogInformation($"Recording Oversight review status of Unsuccessful for gateway fail application Id {request.ApplicationId}");
 
+            var application = await _applyRepository.GetApplication(request.ApplicationId);
+
+            if (application == null)
+            {
+                throw new InvalidOperationException($"Application {request.ApplicationId} not found");
```

Comment:
Is it worth putting a Log entry here? Does the exception get handled somewhere else?